components/scream/src/share/scream_logging.hpp

```diff
@@ -0,0 +1,271 @@
+
+#ifndef SCREAM_LOGGING_HPP
+#define SCREAM_LOGGING_HPP
+
+#include <cstdio>
+#include <cstdlib>
+#include <string>
+#include <vector>
+#include <stdexcept>
+#include "scream_config.h"
+
+#if LOG_STACK_TRACE
+#include <execinfo.h>
+#endif
+
+namespace scream {
+namespace util {
+struct Error : public std::runtime_error {
```

Comment:
Does this struct have to come before the #includes below?

Reply:
Not necessary, we could change it.